aeon/dj_pipeline/analysis/visit_analysis.py

```diff
@@ -388,6 +388,86 @@ def make(self, key):
             self.Nest.insert(in_nest_times)
             self.FoodPatch.insert(in_food_patch_times)
 
+    @classmethod
+    def _get_foraging_bouts(
+        cls,
+        visit_per_day_row,
+        wheel_dist_crit=None,
+        min_bout_duration=None,
+        using_aeon_io=False,
+    ):
+
+        # Get number of foraging bouts
+        nb_bouts = 0
+
+        in_patch = visit_per_day_row["in_patch"]
+        if np.size(in_patch) == 0:  # no food patch position timestamps
+            return nb_bouts
+
+        change_ind = (
+            np.where((np.diff(in_patch) / 1e6) > np.timedelta64(20))[0] + 1
+        )  # timestamp index where state changes
+
+        # print(
+        #     visit_per_day_row["subject"],
+        #     visit_per_day_row["visit_date"],
+        #     visit_per_day_row["food_patch_description"],
+        # )
+
+        if np.size(change_ind) == 0:  # one contiguous block
+
+            wheel_start, wheel_end = in_patch[0], in_patch[-1]
+            ts_duration = (wheel_end - wheel_start) / np.timedelta64(
+                1, "s"
+            )  # in seconds
+            if ts_duration < min_bout_duration:
+                return nb_bouts
+
+            wheel_data = acquisition.FoodPatchWheel.get_wheel_data(
+                experiment_name="exp0.2-r0",
+                start=wheel_start,
+                end=wheel_end,
+                patch_name=visit_per_day_row["food_patch_description"],
+                using_aeon_io=using_aeon_io,
+            )
+            if wheel_data.distance_travelled[-1] > wheel_dist_crit:
+                return nb_bouts + 1
+            else:
+                return nb_bouts
+
+        # fetch contiguous timestamp blocks
+        for i in range(len(change_ind) + 1):
+            if i == 0:
+                ts_array = in_patch[: change_ind[i]]
+            elif i == len(change_ind):
+                ts_array = in_patch[change_ind[i - 1] :]
+            else:
+                ts_array = in_patch[change_ind[i - 1] : change_ind[i]]
+
+            ts_duration = (ts_array[-1] - ts_array[0]) / np.timedelta64(
+                1, "s"
+            )  # in seconds
+            if ts_duration < min_bout_duration:
+                continue
+
+            wheel_start, wheel_end = ts_array[0], ts_array[-1]
+            if wheel_start > wheel_end:  # skip if timestamps were misaligned
+                continue
+
+            wheel_data = acquisition.FoodPatchWheel.get_wheel_data(
+                experiment_name="exp0.2-r0",
```

Comment:
why hardcoding `experiment_name` here? can we retrieve it from the input (e.g. if we take a `visit_day_key` as input, we can get `experiment_name` from there)